### PR TITLE
NAS-118855 / 22.12 / Dump currently active WebSocket sessions. Allow terminating all excep…

### PR DESCRIPTION
**Pull request by themylogin**
…t the current one.

### CODE REVIEW

**Review by yocalebo — COMMENTED** [review]
Does this tear down the session between the 2 controllers in an HA system?

**Comment by themylogin** [comment]
@yocalebo no, it checks for `is_internal_session` and those are `True` for HA connection

File: src/middlewared/middlewared/plugins/auth.py [hunk 1]
```diff
@@ -264,6 +281,40 @@ def sessions(self, app, filters, options):
             options,
         )
 
+    @accepts(Str('id'))
+    @returns(Bool(description='Is `true` if `session was terminated successfully'))
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/auth.py:285]: There is an extra quote before session word

**Comment by william-gr** [comment]
@truenas/middleware can anyone else please review this, please?

Looks good at first glance

File: src/middlewared/middlewared/plugins/auth.py [hunk 2]
```diff
@@ -142,22 +142,30 @@ def _app_on_close(self, app):
         self.logout(app)
 
 
+def dump_credentials(credentials):
+    return {
+        "credentials": re.sub(
+            "([A-Z])",
+            "_\\1",
```
Comment (sonicaj) [inline src/middlewared/middlewared/plugins/auth.py:149]: Perhaps let's use raw strings ?

File: src/middlewared/middlewared/plugins/auth.py [hunk 3]
```diff
@@ -264,6 +281,40 @@ def sessions(self, app, filters, options):
...
+        """
+        session = self.session_manager.sessions.get(id)
+        if session is None:
+            return False
+
+        await session.app.response.close()
+
+    @accepts()
+    @pass_app()
+    async def terminate_other_sessions(self, app):
```
Comment (sonicaj) [inline src/middlewared/middlewared/plugins/auth.py:298]: Missing returns decorator ?

**Review by sonicaj — APPROVED** [review]

**Pull request merged**